public/app/features/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocation.utils.ts

```diff
@@ -0,0 +1,65 @@
+import { StorageLocation, LocationType, S3Location } from '../StorageLocations.types';
+import { AddStorageLocationFormProps } from './AddStorageLocationModal.types';
+
+export const toStorageLocation = (values: AddStorageLocationFormProps): StorageLocation => {
+  const { name, description, type, endpoint, client, server, accessKey, secretKey } = values;
+  const result: Partial<StorageLocation> = { name, description };
+
+  switch (type) {
+    case LocationType.S3:
+      result.type = LocationType.S3;
```

Comment:
Don't need to set the type in every case since it doesn't change.

Reply:
yep